### ISSUE TITLE
xUnit 2.3.0 VS Runner breaks Live Unit Testing feature of VS 2017

### ISSUE THREAD

**Issue by kstreith** [xunit/xunit#1514]
Repo project here:
https://github.com/kstreith/xunit23liveunittestbug

If you use the xunit.runner.visualstudio 2.3.0 package, live unit testing is broken. Open the ConsoleApp1.sln in LiveTestsBrokenWith_xUnit23Vs23Runner in VS 2017. Enable live unit testing. It will fail with the following messages in the Output window (Live Unit Testing output).

```
[17:30:19.495 Info] Live Unit Testing started.
[17:30:22.483 Info] Build completed (succeeded).
[17:30:27.009 Info] Started executing 1 tests.
[17:30:28.862 Info] [TestRunner 1] [xUnit.net 00:00:00.3146146] UnitTestProject1: Catastrophic error during deserialization: System.ArgumentException: Could not load type 
Parameter name: serializedValue
   at Xunit.Sdk.SerializationHelper.Deserialize[T](String serializedValue)
   at Xunit.Sdk.TestFrameworkExecutor`1.Deserialize(String value)
   at Xunit.DefaultTestCaseBulkDeserializer.<BulkDeserialize>b__2_0(String serialization)
   at System.Linq.Enumerable.<>c__DisplayClass7_0`3.<CombineSelectors>b__0(TSource x)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Xunit.DefaultTestCaseBulkDeserializer.BulkDeserialize(List`1 serializations)
   at Xunit.Xunit2.BulkDeserialize(List`1 serializations)
   at Xunit.Runner.VisualStudio.VsTestRunner.RunTestsInAssembly(IRunContext runContext, IFrameworkHandle frameworkHandle, LoggerHelper logger, TestPlatformContext testPlatformContext, IMessageSinkWithTypes reporterMessageHandler, AssemblyRunInfo runInfo)
```

This is only an issue with the xunit.runner.visualstudio 2.3.0 package. If you look in the repo solution directory, I have another project. The LiveTestsWorksWith_xUnit23Vs22Runner project works correctly with Live Unit Tests, it has xunit 2.3.0 but uses the earlier version of the visualstudio runner, e.g. 2.2.0.

I used Visual Studio 15.3.3. Details on Live Unit Testing of Visual Studio: https://docs.microsoft.com/en-us/visualstudio/test/live-unit-testing

I'd like to be able to use the new visualstudio runner with live unit testing feature of VS.

**Comment by bradwilson** [comment]
```
Xunit.DefaultTestCaseBulkDeserializer.BulkDeserialize
```

This is not a good sign. ☹️

**Comment by felipeOliveira** [comment]
I have same problem here =(

**Comment by smadala** [comment]
To unblock yourself, Downgrade xunit packages to 2.3.0-beta5-* which may cause performance degrade.

**Comment by campersau** [comment]
We have used xunit 2.2.0 but after the Windows Fall Creator Update 2017 yesterday we got the following error:
```
Catastrophic failure: System.TypeInitializationException: Der Typeninitialisierer für "Xunit.DiaSession" hat eine Ausnahme verursacht. ---> System.IO.FileNotFoundException: Die Datei oder Assembly "System.Reflection.TypeExtensions, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
```

So we updated to xunit 2.3.0 and there we had the same problem as described in the bug report.
Even though we don't do live unit tests but denying app domains.

Downgrading to xunit 2.3.0-beta5-* resolved all issues.

**Comment by paulomorgado** [comment]
On the other hand, using the regular Visual Studio test runner yields the same error with xunit 2.3.0-beta5-build3769.

**Comment by majorimi** [comment]
I'm using xUnit 2.3.1, xunit.runner.visualstudio 2.3.1 with VS 2017 Enterprise 15.7.5
Live Unit testing still does not work...

**Comment by bradwilson** [comment]
@majorimi Does it still repro with 2.4 RC 1?

**Comment by bradwilson** [comment]
(if so, please open a new issue with a repro)

**Comment by majorimi** [comment]
Thanks for the reply @bradwilson. I'm not a big fan of using Pre-Release packages because they tend to break something. Since the feature is already broken I tried your suggestion. Even used a newer version than you suggested: 2.4.0-rc.2.build4045. Result is the same... Live unit testing does not work.

Looking into the Output window I just got a very weird error message:

> [08:57:44.028 Info] Live Unit Testing started.
> **[08:57:45.001 Info] Build completed (failed).**

**But it is completely bogus. I can build, run, debug the project.** So there is no build error. Additional info: my project is a Visual Studio Extension. Can it be the root cause of the issue?
